cmd/common/image/inspect.go

```diff
@@ -107,35 +109,69 @@ func NewInspectCmd(runtime runtime.Runtime) *cobra.Command {
 				"content":     string(v.Content),
 			}
 		}
+
+		extraInfo, _ := cmd.PersistentFlags().GetString("extra-info")
+		if extraInfo != "" {
+			if _, ok := info.ExtraInfo.Data[extraInfo]; !ok {
+				return fmt.Errorf("extra info %q not found", extraInfo)
+			}
+
+			jsonPath, _ := cmd.PersistentFlags().GetString("jsonpath")
+
+			dataEntry := info.ExtraInfo.Data[extraInfo]
+			customMap := make(map[string]interface{})
+			if dataEntry.ContentType == "application/json" {
+				// unmarshal the JSON content
+				var dataEntryContent interface{}
+				if err := json.Unmarshal(dataEntry.Content, &dataEntryContent); err != nil {
+					return fmt.Errorf("unmarshalling JSON content: %w", err)
+				}
+				customMap[extraInfo] = dataEntryContent
+			} else {
+				customMap[extraInfo] = string(dataEntry.Content)
+			}
+			extraInfoMap = customMap
+			if jsonPath != "" {
+				// unmarshal the JSON content
+				var dataEntryContent interface{}
+				if err := json.Unmarshal(dataEntry.Content, &dataEntryContent); err != nil {
+					return fmt.Errorf("unmarshalling JSON content: %w", err)
+				}
+
+				result, err := jsonpath.Get(fmt.Sprintf("$%s", jsonPath), dataEntryContent)
+				if err != nil {
+					return fmt.Errorf("resolving path %q: %w", jsonPath, err)
+				}
+				if result == nil {
+					return fmt.Errorf("path %q not found in extra info %q", jsonPath, extraInfo)
+				}
+
+				// overwrite extrainfo map
+				customMap := make(map[string]interface{})
+				customMap[fmt.Sprintf("%s%s", extraInfo, jsonPath)] = result
+				extraInfoMap = customMap
```

Comment:
> I refactored and removed the custom output mode, but in doing so our output format broke too, here's an example:
> 
> ```
> $ sudo ./ig image inspect fsnotify --extra-info ebpf.programs --jsonpath='[*].Section' -o jsonpretty
> {
>   "ebpf.programs[*].Section": [
>     "kprobe/inotify_handle_event",
> ```
> [...]

This is the code that makes the key like "ebpf.programs[*].Section". I think you can keep the variable "result", and use it in the `switch outputMode` below.